src/main/scala/cromwell/engine/db/LiquibaseUtils.scala

```diff
@@ -0,0 +1,121 @@
+    */
+  def updateSchema(jdbcConnection: Connection): Unit = {
+    val liquibaseConnection = newConnection(jdbcConnection)
+    try {
+      val liquibase = new Liquibase(DefaultChangeLog, new ClassLoaderResourceAccessor(), liquibaseConnection)
+      checkForChangeLogDir(liquibase)
+      updateSchema(liquibase)
+    } finally {
+      closeConnection(liquibaseConnection)
+    }
+  }
+
+  /**
+    * Wraps a jdbc connection in the database with the appropriate liquibase connection.
+    * As of 3.4.x, liquibase uses a custom connection for Hsql, Sybase, and Derby, although only Hsql is supported by
+    * cromwell.
+    *
+    * @param jdbcConnection The liquibase connection.
+    * @return
+    */
+  def newConnection(jdbcConnection: Connection): DatabaseConnection = {
+    jdbcConnection.getMetaData.getDatabaseProductName match {
+      case HsqlDatabaseProperties.PRODUCT_NAME => new HsqlConnection(jdbcConnection)
+      case _ => new JdbcConnection(jdbcConnection)
+    }
+  }
+
+  /**
+    * Looks for evidence that the change log was previously applied using files in a directory under
+    * src/main/migrations/. If found, throws an error stating that the database must be manually updated first to
+    * remove the paths from the database. The sql returned in the error message updates the file paths, and also resets
+    * the md5s to null such that they'll be recalculated on the next update.
+    *
+    * Good example:
+    *
+    *   mysql> select distinct filename from databasechangelog limit 3;
+    *   +--------------------------------------+
+    *   | filename                             |
+    *   +--------------------------------------+
+    *   | changesets/db_schema.xml             |
+    *   | changesets/symbol_iteration_null.xml |
+    *   | changesets/wdl_and_inputs.xml        |
+    *   +--------------------------------------+
+    *   3 rows in set (0.00 sec)
+    *
+    * Example that needs an update:
+    *
+    *   mysql> select distinct filename from databasechangelog limit 3;
+    *   +---------------------------------------------------------------------------------------------------+
+    *   | filename                                                                                          |
+    *   +---------------------------------------------------------------------------------------------------+
+    *   | /root/github.com/broadinstitute/cromwell/src/main/migrations/changesets/db_schema.xml             |
+    *   | /root/github.com/broadinstitute/cromwell/src/main/migrations/changesets/symbol_iteration_null.xml |
+    *   | /root/github.com/broadinstitute/cromwell/src/main/migrations/changesets/wdl_and_inputs.xml        |
+    *   +---------------------------------------------------------------------------------------------------+
+    *   3 rows in set (0.00 sec)
+    *
+    * @param liquibase The facade for interacting with liquibase.
+    */
+  def checkForChangeLogDir(liquibase: Liquibase): Unit = {
```

Comment:
My understanding of this function is that it's for migrations from the current database scheme and won't be needed once all DB schemes are updated.. is that right?